src/Skybrud.Umbraco.BorgerDk/Caching/BorgerDkCachingService.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using NPoco;
+using Skybrud.Umbraco.BorgerDk.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Cms.Core.Scoping;
+using Umbraco.Cms.Infrastructure.Persistence;
+using Umbraco.Extensions;
+
+namespace Skybrud.Umbraco.BorgerDk.Caching {
+    public class BorgerDkCachingService {
+        private readonly IMemoryCache cache;
+        private readonly IScopeProvider scopeProvider;
+        private readonly ILogger<BorgerDkCachingService> logger;
+        private readonly MemoryCacheEntryOptions cacheEntryOptions;
+        public List<string> CurrentlyCachedIds { get; set; } = new List<string>();
+
+        public BorgerDkCachingService(IMemoryCache cache, IScopeProvider scopeProvider, ILogger<BorgerDkCachingService> logger) {
+            this.cache = cache;
+            this.scopeProvider = scopeProvider;
+            this.logger = logger;
+            cacheEntryOptions = new MemoryCacheEntryOptions()
+                .SetAbsoluteExpiration(TimeSpan.FromDays(1));
+
+            RefreshCache();
+        }
+
+        public void CacheArticle(BorgerDkArticleDto article) {
+            cache.Set(article.Id, article, cacheEntryOptions);
+            CurrentlyCachedIds.Add(article.Id);
```

Comment:
The implementation around `CurrentlyCachedIds` seems a bit fishy. I think ideally you should use a dictionary instead for storing the articles, and then you might not need to use the memory cache.